File: Microsoft/EvaluateReversePolishNotation.cpp
class Solution
{
public:
    int evalRPN(vector<string> &tokens)
    {
        stack<int> s;
        for (auto x : tokens)
        {
            if (x == "+" || x == "-" || x == "/" || x == "*")
            {
                int op2 = s.top();
                s.pop();
                int op1 = s.top();
                s.pop();

                if (x == "+")
                {
                    s.push(op1 + op2);
                }
                if (x == "-")
                {
                    s.push(op1 - op2);
                }
                if (x == "/")
                {
                    s.push(op1 / op2);
                }
                if (x == "*")
                {
                    s.push(op1 * op2);
                }
            }
            else
            {
                stringstream ss(x);
                int d;
                ss >> d;
                s.push(d);
            }
        }
        return s.top();
    }
};

File: Microsoft/EvaluateReversePolishNotation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stk = []
        for x in tokens:
            if x not in ["+","-","/","*"]:
                stk.append(int(x))
            else:
                a = stk.pop(-1)
                b = stk.pop(-1)
                if x == "/":
                    stk.append(int(b/a))
                else:
                    stk.append(eval(f"{b}{x}{a}"))

        return stk[0]